src/containers/EventsList.js

```diff
@@ -1,18 +1,26 @@
 import React from "react";
 import { connect } from "react-redux";
-import EventItem from "./EventItem";
+import PropTypes from "prop-types";
+
+import EventItem from "../components/EventItem";
 import visibleEvents from "../selectors/visibleEvents";
 
 const EventsList = ({ events }) => (
   <>
-    {events.map(event => (
-      <EventItem key={event.id} {...event} />
-    ))}
+    {events.length === 0 ? (
+      <p>No events</p>
+    ) : (
+      events.map(event => <EventItem key={event.id} {...event} />)
+    )}
   </>
 );
 
 const mapStateToProps = state => ({
   events: visibleEvents(state.events, state.filters),
 });
 
+EventsList.propTypes = {
+  events: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
If you create a separate object for event prop types, then maybe you would be able to use it also here, instead of plain `PropTypes.object` 🤔